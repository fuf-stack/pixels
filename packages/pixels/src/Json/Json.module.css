.wrapper {
  :global {
    .variable-row .copy-to-clipboard-container,
    .object-meta-data .copy-to-clipboard-container {
      display: inline-block;
      visibility: hidden;
    }

    .object-meta-data:hover,
    .variable-row:hover,
    .object-key-val > span:hover {
      .copy-to-clipboard-container {
        visibility: visible;
      }
    }
  }
}
