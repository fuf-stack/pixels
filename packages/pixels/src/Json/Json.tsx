import type { ReactNode } from 'react';

import { useState } from 'react';
import { FaChevronDown, FaChevronUp, FaTimesCircle } from 'react-icons/fa';
import ReactJson from 'react-json-view';

import cn from 'classnames';

import Button from '../Button';

import styles from './Json.module.css';

/**
 * returns the value (JSON string or object) as object
 */
const getValue = (value: string | object) => {
  if (typeof value === 'string') {
    return JSON.parse(value);
  }
  if (typeof value === 'object') {
    return value;
  }
  throw new Error(`${typeof value} can not be visualized`);
};

export interface JsonProps {
  /** CSS class name */
  className?: string | string[] | null;
  /** When set to true, all nodes will be collapsed by default. Use an integer value to collapse at a particular depth. */
  collapsed?: boolean | number;
  /** Object to be visualized JSON string or object */
  value: string | object;
}

/**
 * Json renderer based on [react-json-view](https://mac-s-g.github.io/react-json-view/demo/dist/)
 */
const Json = ({ className = null, collapsed = false, value }: JsonProps) => {
  const isDarkMode = document.body.classList.contains('dark');

  let content: ReactNode = null;
  let error: ReactNode = null;

  const [showDetails, setShowDetails] = useState(false);

  try {
    content = (
      <ReactJson
        theme={isDarkMode ? 'tomorrow' : 'rjv-default'}
        style={{
          fontSize: '12px',
          backgroundColor: 'unset',
        }}
        name={false}
        displayDataTypes={false}
        src={getValue(value)}
        collapsed={collapsed}
      />
    );
  } catch (err) {
    error = (
      <div
        className="mb-4 flex flex-col items-center rounded-lg border border-danger bg-danger-50 p-4 text-sm text-danger"
        role="alert"
      >
        <div className="flex w-full justify-between gap-6">
          <div className="flex items-center">
            <FaTimesCircle className="mr-2" />
            <span className="font-medium">Failed to parse JSON data</span>
          </div>
          <Button
            color="danger"
            size="sm"
            variant="light"
            onClick={() => setShowDetails(!showDetails)}
          >
            {showDetails ? (
              <>
                <FaChevronUp /> Hide Details
              </>
            ) : (
              <>
                <FaChevronDown /> Show Details
              </>
            )}
          </Button>
        </div>
        {showDetails && (
          <div className="mt-4 w-full text-left">
            <div>
              <strong>Error:</strong>
              <pre>
                {/* @ts-expect-error is ok */}
                {err?.name}: {err?.message}
              </pre>
            </div>
            <div className="mt-4">
              <strong>Data:</strong>
              <pre>
                {typeof value !== 'string'
                  ? JSON.stringify(value, null, 2)
                  : value}
              </pre>
            </div>
          </div>
        )}
      </div>
    );
  }
  return (
    <div className={cn(styles.wrapper, className)}>{error || content}</div>
  );
};

export default Json;
