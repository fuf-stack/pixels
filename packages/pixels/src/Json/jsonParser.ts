export class JsonParseError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'JsonParseError';
  }
}

export const getValue = (value: string | object): object => {
  if (typeof value === 'string') {
    try {
      return JSON.parse(value);
    } catch (error) {
      throw new JsonParseError(
        `Invalid JSON string: ${(error as Error).message}`,
      );
    }
  }
  if (value === null || value === undefined) {
    throw new JsonParseError('Value cannot be null or undefined');
  }
  if (typeof value === 'object') {
    return value;
  }
  throw new JsonParseError(`${typeof value} cannot be visualized`);
};
