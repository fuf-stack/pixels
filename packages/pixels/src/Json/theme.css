/********************************************************************
 * light theme variables
  * see: https://github.com/uiwjs/react-json-view/blob/main/core/src/theme/light.ts
 ********************************************************************/
 :root {
  --w-rjv-font-family: monospace;
  --w-rjv-color: #002b36;
  --w-rjv-key-string: #002b36;
  --w-rjv-background-color: unset;
  --w-rjv-line-color: #ebebeb;
  --w-rjv-arrow-color: var(--w-rjv-color);
  --w-rjv-edit-color: var(--w-rjv-color);
  --w-rjv-info-color: rgba(0, 0, 0, 0.3);
  --w-rjv-update-color: #ebcb8b;
  --w-rjv-copied-color: #002b36;
  --w-rjv-copied-success-color: #28a745;

  --w-rjv-curlybraces-color: #236a7c;
  --w-rjv-colon-color: #002b36;
  --w-rjv-brackets-color: #236a7c;
  --w-rjv-quotes-color: var(--w-rjv-key-string);
  --w-rjv-quotes-string-color: var(--w-rjv-type-string-color);

  --w-rjv-type-string-color: #cb4b16;
  --w-rjv-type-int-color: #268bd2;
  --w-rjv-type-float-color: #859900;
  --w-rjv-type-bigint-color: #268bd2;
  --w-rjv-type-boolean-color: #2aa198;
  --w-rjv-type-date-color: #586e75;
  --w-rjv-type-url-color: #0969da;
  --w-rjv-type-null-color: #d33682;
  --w-rjv-type-nan-color: #859900;
  --w-rjv-type-undefined-color: #586e75;
}

/********************************************************************
 * dark theme variables
 * see: https://github.com/uiwjs/react-json-view/blob/main/core/src/theme/vscode.ts
 ********************************************************************/
:root[data-theme="dark"] {
  --w-rjv-font-family: monospace;
  --w-rjv-color: #9cdcfe;
  --w-rjv-key-string: #9cdcfe;
  --w-rjv-background-color: unset;
  --w-rjv-line-color: rgba(54, 51, 66, 0.5);
  --w-rjv-arrow-color: #838383;
  --w-rjv-edit-color: var(--w-rjv-color);
  --w-rjv-info-color: rgba(156, 156, 156, 0.48);
  --w-rjv-update-color: #9cdcfe;
  --w-rjv-copied-color: #9cdcfe;
  --w-rjv-copied-success-color: #28a745;

  --w-rjv-curlybraces-color: #d4d4d4;
  --w-rjv-colon-color: #d4d4d4;
  --w-rjv-brackets-color: #d4d4d4;
  --w-rjv-quotes-color: var(--w-rjv-key-string);
  --w-rjv-quotes-string-color: var(--w-rjv-type-string-color);

  --w-rjv-type-string-color: #ce9178;
  --w-rjv-type-int-color: #b5cea8;
  --w-rjv-type-float-color: #b5cea8;
  --w-rjv-type-bigint-color: #b5cea8;
  --w-rjv-type-boolean-color: #569cd6;
  --w-rjv-type-date-color: #b5cea8;
  --w-rjv-type-url-color: #3b89cf;
  --w-rjv-type-null-color: #569cd6;
  --w-rjv-type-nan-color: #859900;
  --w-rjv-type-undefined-color: #569cd6;
}
