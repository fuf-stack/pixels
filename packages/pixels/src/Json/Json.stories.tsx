import type { Meta, StoryObj } from '@storybook/react-vite';

import Json from './Json';

const meta: Meta<typeof Json> = {
  title: 'pixels/Json',
  component: Json,
};

export default meta;
type Story = StoryObj<typeof Json>;

const testObject = {
  aList: [1, 2, 3],
  anObject: { bool: true, number: 123, null: null, string: 'a string' },
};

export const WithJsonString: Story = {
  args: {
    value: JSON.stringify(testObject, null, 2),
  },
};

export const WithJsonObject: Story = {
  args: {
    value: testObject,
  },
};

export const Collapsed: Story = {
  args: {
    collapsed: true,
    value: testObject,
  },
};

export const InvalidJson: Story = {
  args: {
    value: 'this is a string, not a Json',
  },
};

const programmersLife = {
  coffee: {
    cups: 42,
    status: 'Not enough',
    lastBreak: "What's a break?",
    type: "Whatever's left in the pot",
  },
  bugs: {
    found: 99,
    fixed: 1,
    created: 157,
    excuses: [
      'It works on my machine',
      "That's not a bug, it's a feature",
      'Cannot reproduce',
      null, // The bug that got away
    ],
  },
  meetings: {
    scheduled: 8,
    necessary: 0,
    timeSpentPretendingToPayAttention: '3 hours',
    cameraMalfunction: true,
    muted: 'Definitely not talking about the project manager',
  },
};

export const DayInTheLifeOfAProgrammer: Story = {
  args: {
    value: programmersLife,
  },
};
