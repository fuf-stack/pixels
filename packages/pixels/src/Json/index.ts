import Json from './Json';

export type { JsonProps } from './Json';

export { Json };

export default Json;
