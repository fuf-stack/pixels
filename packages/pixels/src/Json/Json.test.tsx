import { describe, expect, test } from 'vitest';

import storySnapshots from '@repo/storybook-config/story-snapshots';
import { fireEvent, render } from '@testing-library/react';

import Json from './Json';
import * as stories from './Json.stories';

describe('Story Snapshots', () => {
  storySnapshots(stories);
});

describe('Coverage', () => {
  test('renders Json component error without data', () => {
    const { getByText } = render(<Json />);

    // Find the "Show Details" button
    const showDetailsButton = getByText('Show Details');
    expect(showDetailsButton).toBeInTheDocument();

    // Click the button
    fireEvent.click(showDetailsButton);

    // Check if the error is shown
    const errorMessage = getByText('Error:');
    expect(errorMessage).toBeInTheDocument();
  });

  test('renders Json component error with data', () => {
    const { getByText } = render(<Json value="this is a string, not a Json" />);

    // Find the "Show Details" button
    const showDetailsButton = getByText('Show Details');
    expect(showDetailsButton).toBeInTheDocument();

    // Click the button
    fireEvent.click(showDetailsButton);

    // Check if the error is shown
    const errorMessage = getByText('Error:');
    expect(errorMessage).toBeInTheDocument();
  });

  test('Json component copy icon shows on hover and turns green on click', () => {
    const { getByTestId, getByText } = render(
      <Json value={{ some: 'data' }} />,
    );

    const mouseOverElement = getByText('1 item');

    fireEvent.mouseOver(mouseOverElement);

    const svgSpan = getByTestId('copy-button');

    fireEvent.click(svgSpan.firstChild);

    fireEvent.mouseOver(mouseOverElement);

    expect(svgSpan.firstChild).toHaveClass('text-success');
  });
});
