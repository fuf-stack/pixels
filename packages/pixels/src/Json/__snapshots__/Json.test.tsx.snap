// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Story Snapshots > AsJsonString 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="w-json-view-container w-rjv pr-5 w-rjv-inner"
      style="line-height: 1.4; font-family: var(--w-rjv-font-family, Menlo, monospace); font-size: 13px; --w-rjv-font-family: monospace; --w-rjv-color: #002b36; --w-rjv-key-string: #002b36; --w-rjv-background-color: #ffffff; --w-rjv-line-color: #ebebeb; --w-rjv-arrow-color: var(--w-rjv-color); --w-rjv-edit-color: var(--w-rjv-color); --w-rjv-info-color: #0000004d; --w-rjv-update-color: #ebcb8b; --w-rjv-copied-color: #002b36; --w-rjv-copied-success-color: #28a745; --w-rjv-curlybraces-color: #236a7c; --w-rjv-colon-color: #002b36; --w-rjv-brackets-color: #236a7c; --w-rjv-quotes-color: var(--w-rjv-key-string); --w-rjv-quotes-string-color: var(--w-rjv-type-string-color); --w-rjv-type-string-color: #cb4b16; --w-rjv-type-int-color: #268bd2; --w-rjv-type-float-color: #859900; --w-rjv-type-bigint-color: #268bd2; --w-rjv-type-boolean-color: #2aa198; --w-rjv-type-date-color: #586e75; --w-rjv-type-url-color: #0969da; --w-rjv-type-null-color: #d33682; --w-rjv-type-nan-color: #859900; --w-rjv-type-undefined-color: #586e75;"
    >
      <span
        style="display: inline-flex; align-items: center;"
      >
        <span
          class="w-rjv-arrow"
          style="transform: rotate(0deg); transition: all 0.3s;"
        >
          <svg
            fill="var(--w-rjv-arrow-color, currentColor)"
            style="cursor: pointer; height: 1em; width: 1em; user-select: none; display: inline-flex;"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="w-rjv-curlybraces-start"
        >
          {
        </span>
        <span
          class="w-rjv-object-size"
          style="padding-left: 8px; font-style: italic;"
        >
          2 items
        </span>
      </span>
      <div
        class="w-rjv-wrap"
        style="padding-left: 15px; margin-left: 6px;"
      >
        <div
          class="w-rjv-inner"
        >
          <span
            style="display: inline-flex; align-items: center;"
          >
            <span
              class="w-rjv-arrow"
              style="transform: rotate(0deg); transition: all 0.3s;"
            >
              <svg
                fill="var(--w-rjv-arrow-color, currentColor)"
                style="cursor: pointer; height: 1em; width: 1em; user-select: none; display: inline-flex;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span>
              <span
                class="w-rjv-quotes"
                data-placement="left"
              >
                "
              </span>
              <span
                class="w-rjv-object-key"
              >
                aList
              </span>
              <span
                class="w-rjv-quotes"
                data-placement="right"
              >
                "
              </span>
            </span>
            <span
              class="w-rjv-colon"
              style="margin-left: 0px; margin-right: 2px;"
            >
              :
            </span>
            <span
              class="w-rjv-brackets-start"
            >
              [
            </span>
            <span
              class="w-rjv-object-size"
              style="padding-left: 8px; font-style: italic;"
            >
              3 items
            </span>
          </span>
          <div
            class="w-rjv-wrap"
            style="padding-left: 15px; margin-left: 6px;"
          >
            <div
              class="w-rjv-line"
            >
              <span>
                <span
                  class="w-rjv-object-key"
                >
                  0
                </span>
              </span>
              <span
                class="w-rjv-colon"
                style="margin-left: 0px; margin-right: 2px;"
              >
                :
              </span>
              <span
                class="w-rjv-value"
                data-type="int"
              >
                1
              </span>
            </div>
            <div
              class="w-rjv-line"
            >
              <span>
                <span
                  class="w-rjv-object-key"
                >
                  1
                </span>
              </span>
              <span
                class="w-rjv-colon"
                style="margin-left: 0px; margin-right: 2px;"
              >
                :
              </span>
              <span
                class="w-rjv-value"
                data-type="int"
              >
                2
              </span>
            </div>
            <div
              class="w-rjv-line"
            >
              <span>
                <span
                  class="w-rjv-object-key"
                >
                  2
                </span>
              </span>
              <span
                class="w-rjv-colon"
                style="margin-left: 0px; margin-right: 2px;"
              >
                :
              </span>
              <span
                class="w-rjv-value"
                data-type="int"
              >
                3
              </span>
            </div>
          </div>
          <div
            style="padding-left: 4px;"
          >
            <span
              class="w-rjv-brackets-end"
            >
              ]
            </span>
          </div>
        </div>
        <div
          class="w-rjv-inner"
        >
          <span
            style="display: inline-flex; align-items: center;"
          >
            <span
              class="w-rjv-arrow"
              style="transform: rotate(0deg); transition: all 0.3s;"
            >
              <svg
                fill="var(--w-rjv-arrow-color, currentColor)"
                style="cursor: pointer; height: 1em; width: 1em; user-select: none; display: inline-flex;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span>
              <span
                class="w-rjv-quotes"
                data-placement="left"
              >
                "
              </span>
              <span
                class="w-rjv-object-key"
              >
                anObject
              </span>
              <span
                class="w-rjv-quotes"
                data-placement="right"
              >
                "
              </span>
            </span>
            <span
              class="w-rjv-colon"
              style="margin-left: 0px; margin-right: 2px;"
            >
              :
            </span>
            <span
              class="w-rjv-curlybraces-start"
            >
              {
            </span>
            <span
              class="w-rjv-object-size"
              style="padding-left: 8px; font-style: italic;"
            >
              4 items
            </span>
          </span>
          <div
            class="w-rjv-wrap"
            style="padding-left: 15px; margin-left: 6px;"
          >
            <div
              class="w-rjv-line"
            >
              <span>
                <span
                  class="w-rjv-quotes"
                  data-placement="left"
                >
                  "
                </span>
                <span
                  class="w-rjv-object-key"
                >
                  bool
                </span>
                <span
                  class="w-rjv-quotes"
                  data-placement="right"
                >
                  "
                </span>
              </span>
              <span
                class="w-rjv-colon"
                style="margin-left: 0px; margin-right: 2px;"
              >
                :
              </span>
              <span
                class="w-rjv-value"
                data-type="bool"
              >
                true
              </span>
            </div>
            <div
              class="w-rjv-line"
            >
              <span>
                <span
                  class="w-rjv-quotes"
                  data-placement="left"
                >
                  "
                </span>
                <span
                  class="w-rjv-object-key"
                >
                  number
                </span>
                <span
                  class="w-rjv-quotes"
                  data-placement="right"
                >
                  "
                </span>
              </span>
              <span
                class="w-rjv-colon"
                style="margin-left: 0px; margin-right: 2px;"
              >
                :
              </span>
              <span
                class="w-rjv-value"
                data-type="int"
              >
                123
              </span>
            </div>
            <div
              class="w-rjv-line"
            >
              <span>
                <span
                  class="w-rjv-quotes"
                  data-placement="left"
                >
                  "
                </span>
                <span
                  class="w-rjv-object-key"
                >
                  null
                </span>
                <span
                  class="w-rjv-quotes"
                  data-placement="right"
                >
                  "
                </span>
              </span>
              <span
                class="w-rjv-colon"
                style="margin-left: 0px; margin-right: 2px;"
              >
                :
              </span>
            </div>
            <div
              class="w-rjv-line"
            >
              <span>
                <span
                  class="w-rjv-quotes"
                  data-placement="left"
                >
                  "
                </span>
                <span
                  class="w-rjv-object-key"
                >
                  string
                </span>
                <span
                  class="w-rjv-quotes"
                  data-placement="right"
                >
                  "
                </span>
              </span>
              <span
                class="w-rjv-colon"
                style="margin-left: 0px; margin-right: 2px;"
              >
                :
              </span>
              <span
                class="w-rjv-quotes"
              >
                "
              </span>
              <span
                class="w-rjv-value"
                data-type="string"
                style="cursor: initial;"
              >
                a string
              </span>
              <span
                class="w-rjv-quotes"
              >
                "
              </span>
            </div>
          </div>
          <div
            style="padding-left: 4px;"
          >
            <span
              class="w-rjv-curlybraces-end"
            >
              }
            </span>
          </div>
        </div>
      </div>
      <div
        style="padding-left: 4px;"
      >
        <span
          class="w-rjv-curlybraces-end"
        >
          }
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Story Snapshots > AsObject 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="w-json-view-container w-rjv pr-5 w-rjv-inner"
      style="line-height: 1.4; font-family: var(--w-rjv-font-family, Menlo, monospace); font-size: 13px; --w-rjv-font-family: monospace; --w-rjv-color: #002b36; --w-rjv-key-string: #002b36; --w-rjv-background-color: #ffffff; --w-rjv-line-color: #ebebeb; --w-rjv-arrow-color: var(--w-rjv-color); --w-rjv-edit-color: var(--w-rjv-color); --w-rjv-info-color: #0000004d; --w-rjv-update-color: #ebcb8b; --w-rjv-copied-color: #002b36; --w-rjv-copied-success-color: #28a745; --w-rjv-curlybraces-color: #236a7c; --w-rjv-colon-color: #002b36; --w-rjv-brackets-color: #236a7c; --w-rjv-quotes-color: var(--w-rjv-key-string); --w-rjv-quotes-string-color: var(--w-rjv-type-string-color); --w-rjv-type-string-color: #cb4b16; --w-rjv-type-int-color: #268bd2; --w-rjv-type-float-color: #859900; --w-rjv-type-bigint-color: #268bd2; --w-rjv-type-boolean-color: #2aa198; --w-rjv-type-date-color: #586e75; --w-rjv-type-url-color: #0969da; --w-rjv-type-null-color: #d33682; --w-rjv-type-nan-color: #859900; --w-rjv-type-undefined-color: #586e75;"
    >
      <span
        style="display: inline-flex; align-items: center;"
      >
        <span
          class="w-rjv-arrow"
          style="transform: rotate(0deg); transition: all 0.3s;"
        >
          <svg
            fill="var(--w-rjv-arrow-color, currentColor)"
            style="cursor: pointer; height: 1em; width: 1em; user-select: none; display: inline-flex;"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="w-rjv-curlybraces-start"
        >
          {
        </span>
        <span
          class="w-rjv-object-size"
          style="padding-left: 8px; font-style: italic;"
        >
          2 items
        </span>
      </span>
      <div
        class="w-rjv-wrap"
        style="padding-left: 15px; margin-left: 6px;"
      >
        <div
          class="w-rjv-inner"
        >
          <span
            style="display: inline-flex; align-items: center;"
          >
            <span
              class="w-rjv-arrow"
              style="transform: rotate(0deg); transition: all 0.3s;"
            >
              <svg
                fill="var(--w-rjv-arrow-color, currentColor)"
                style="cursor: pointer; height: 1em; width: 1em; user-select: none; display: inline-flex;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span>
              <span
                class="w-rjv-quotes"
                data-placement="left"
              >
                "
              </span>
              <span
                class="w-rjv-object-key"
              >
                aList
              </span>
              <span
                class="w-rjv-quotes"
                data-placement="right"
              >
                "
              </span>
            </span>
            <span
              class="w-rjv-colon"
              style="margin-left: 0px; margin-right: 2px;"
            >
              :
            </span>
            <span
              class="w-rjv-brackets-start"
            >
              [
            </span>
            <span
              class="w-rjv-object-size"
              style="padding-left: 8px; font-style: italic;"
            >
              3 items
            </span>
          </span>
          <div
            class="w-rjv-wrap"
            style="padding-left: 15px; margin-left: 6px;"
          >
            <div
              class="w-rjv-line"
            >
              <span>
                <span
                  class="w-rjv-object-key"
                >
                  0
                </span>
              </span>
              <span
                class="w-rjv-colon"
                style="margin-left: 0px; margin-right: 2px;"
              >
                :
              </span>
              <span
                class="w-rjv-value"
                data-type="int"
              >
                1
              </span>
            </div>
            <div
              class="w-rjv-line"
            >
              <span>
                <span
                  class="w-rjv-object-key"
                >
                  1
                </span>
              </span>
              <span
                class="w-rjv-colon"
                style="margin-left: 0px; margin-right: 2px;"
              >
                :
              </span>
              <span
                class="w-rjv-value"
                data-type="int"
              >
                2
              </span>
            </div>
            <div
              class="w-rjv-line"
            >
              <span>
                <span
                  class="w-rjv-object-key"
                >
                  2
                </span>
              </span>
              <span
                class="w-rjv-colon"
                style="margin-left: 0px; margin-right: 2px;"
              >
                :
              </span>
              <span
                class="w-rjv-value"
                data-type="int"
              >
                3
              </span>
            </div>
          </div>
          <div
            style="padding-left: 4px;"
          >
            <span
              class="w-rjv-brackets-end"
            >
              ]
            </span>
          </div>
        </div>
        <div
          class="w-rjv-inner"
        >
          <span
            style="display: inline-flex; align-items: center;"
          >
            <span
              class="w-rjv-arrow"
              style="transform: rotate(0deg); transition: all 0.3s;"
            >
              <svg
                fill="var(--w-rjv-arrow-color, currentColor)"
                style="cursor: pointer; height: 1em; width: 1em; user-select: none; display: inline-flex;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span>
              <span
                class="w-rjv-quotes"
                data-placement="left"
              >
                "
              </span>
              <span
                class="w-rjv-object-key"
              >
                anObject
              </span>
              <span
                class="w-rjv-quotes"
                data-placement="right"
              >
                "
              </span>
            </span>
            <span
              class="w-rjv-colon"
              style="margin-left: 0px; margin-right: 2px;"
            >
              :
            </span>
            <span
              class="w-rjv-curlybraces-start"
            >
              {
            </span>
            <span
              class="w-rjv-object-size"
              style="padding-left: 8px; font-style: italic;"
            >
              4 items
            </span>
          </span>
          <div
            class="w-rjv-wrap"
            style="padding-left: 15px; margin-left: 6px;"
          >
            <div
              class="w-rjv-line"
            >
              <span>
                <span
                  class="w-rjv-quotes"
                  data-placement="left"
                >
                  "
                </span>
                <span
                  class="w-rjv-object-key"
                >
                  bool
                </span>
                <span
                  class="w-rjv-quotes"
                  data-placement="right"
                >
                  "
                </span>
              </span>
              <span
                class="w-rjv-colon"
                style="margin-left: 0px; margin-right: 2px;"
              >
                :
              </span>
              <span
                class="w-rjv-value"
                data-type="bool"
              >
                true
              </span>
            </div>
            <div
              class="w-rjv-line"
            >
              <span>
                <span
                  class="w-rjv-quotes"
                  data-placement="left"
                >
                  "
                </span>
                <span
                  class="w-rjv-object-key"
                >
                  number
                </span>
                <span
                  class="w-rjv-quotes"
                  data-placement="right"
                >
                  "
                </span>
              </span>
              <span
                class="w-rjv-colon"
                style="margin-left: 0px; margin-right: 2px;"
              >
                :
              </span>
              <span
                class="w-rjv-value"
                data-type="int"
              >
                123
              </span>
            </div>
            <div
              class="w-rjv-line"
            >
              <span>
                <span
                  class="w-rjv-quotes"
                  data-placement="left"
                >
                  "
                </span>
                <span
                  class="w-rjv-object-key"
                >
                  null
                </span>
                <span
                  class="w-rjv-quotes"
                  data-placement="right"
                >
                  "
                </span>
              </span>
              <span
                class="w-rjv-colon"
                style="margin-left: 0px; margin-right: 2px;"
              >
                :
              </span>
            </div>
            <div
              class="w-rjv-line"
            >
              <span>
                <span
                  class="w-rjv-quotes"
                  data-placement="left"
                >
                  "
                </span>
                <span
                  class="w-rjv-object-key"
                >
                  string
                </span>
                <span
                  class="w-rjv-quotes"
                  data-placement="right"
                >
                  "
                </span>
              </span>
              <span
                class="w-rjv-colon"
                style="margin-left: 0px; margin-right: 2px;"
              >
                :
              </span>
              <span
                class="w-rjv-quotes"
              >
                "
              </span>
              <span
                class="w-rjv-value"
                data-type="string"
                style="cursor: initial;"
              >
                a string
              </span>
              <span
                class="w-rjv-quotes"
              >
                "
              </span>
            </div>
          </div>
          <div
            style="padding-left: 4px;"
          >
            <span
              class="w-rjv-curlybraces-end"
            >
              }
            </span>
          </div>
        </div>
      </div>
      <div
        style="padding-left: 4px;"
      >
        <span
          class="w-rjv-curlybraces-end"
        >
          }
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Story Snapshots > DefaultCollapsed 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="w-json-view-container w-rjv pr-5 w-rjv-inner"
      style="line-height: 1.4; font-family: var(--w-rjv-font-family, Menlo, monospace); font-size: 13px; --w-rjv-font-family: monospace; --w-rjv-color: #002b36; --w-rjv-key-string: #002b36; --w-rjv-background-color: #ffffff; --w-rjv-line-color: #ebebeb; --w-rjv-arrow-color: var(--w-rjv-color); --w-rjv-edit-color: var(--w-rjv-color); --w-rjv-info-color: #0000004d; --w-rjv-update-color: #ebcb8b; --w-rjv-copied-color: #002b36; --w-rjv-copied-success-color: #28a745; --w-rjv-curlybraces-color: #236a7c; --w-rjv-colon-color: #002b36; --w-rjv-brackets-color: #236a7c; --w-rjv-quotes-color: var(--w-rjv-key-string); --w-rjv-quotes-string-color: var(--w-rjv-type-string-color); --w-rjv-type-string-color: #cb4b16; --w-rjv-type-int-color: #268bd2; --w-rjv-type-float-color: #859900; --w-rjv-type-bigint-color: #268bd2; --w-rjv-type-boolean-color: #2aa198; --w-rjv-type-date-color: #586e75; --w-rjv-type-url-color: #0969da; --w-rjv-type-null-color: #d33682; --w-rjv-type-nan-color: #859900; --w-rjv-type-undefined-color: #586e75;"
    >
      <span
        style="display: inline-flex; align-items: center;"
      >
        <span
          class="w-rjv-arrow"
          style="transform: rotate(-90deg); transition: all 0.3s;"
        >
          <svg
            fill="var(--w-rjv-arrow-color, currentColor)"
            style="cursor: pointer; height: 1em; width: 1em; user-select: none; display: inline-flex;"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="w-rjv-curlybraces-start"
        >
          {
        </span>
        <span
          class="w-rjv-ellipsis"
          style="cursor: pointer; user-select: none;"
        >
          ...
        </span>
        <span
          class="w-rjv-curlybraces-end"
        >
          }
        </span>
        <span
          class="w-rjv-object-size"
          style="padding-left: 8px; font-style: italic;"
        >
          2 items
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`Story Snapshots > InvalidJson 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="mb-4 flex flex-col items-center rounded-lg border border-danger bg-danger-50 p-4 text-sm text-danger"
      role="alert"
    >
      <div
        class="flex w-full justify-between gap-6"
      >
        <div
          class="flex items-center"
        >
          <svg
            class="mr-2"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
            />
          </svg>
          <span
            class="font-medium"
          >
            Failed to parse JSON data
          </span>
        </div>
        <button
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-3 min-w-16 h-8 text-tiny gap-2 rounded-small [&>svg]:max-w-[theme(spacing.8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none bg-transparent text-danger data-[hover=true]:bg-danger/20"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            />
          </svg>
           Show Details
        </button>
      </div>
    </div>
  </div>
</div>
`;
